record-benchmarks.py

```diff
@@ -25,17 +71,36 @@ def pytest_results_to_df(path, run_date):
     df = df.drop(df.index[0])
     return df
 
-# convert bytes to biggest denomination
-def convert_size(size_bytes):
-   if size_bytes == 0:
-       return "0B"
-   size_name = ("B", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB")
-   i = int(math.floor(math.log(size_bytes, 1024)))
-   p = math.pow(1024, i)
-   s = round(size_bytes / p, 2)
-   return "%s %s" % (s, size_name[i])
+def _convert_size(size_bytes):
+    """
+    Convert bytes to biggest denomination.
+    
+    Parameters:
+    size_bytes (str): the number of bytes to be converted.
+
+    Returns:
+    (str): the properly rounded size denomination.
+    """
+    if size_bytes == 0:
+        return "0B"
+    size_name = ("B", "KB", "MB", "GB", "TB", "PB", "EB", "ZB", "YB")
+    i = int(math.floor(math.log(size_bytes, 1024)))
+    p = math.pow(1024, i)
+    s = round(size_bytes / p, 2)
+    return "%s %s" % (s, size_name[i])
+
+def _group_integers_into_ranges(lst):
+    ranges = []
+    for k, g in groupby(enumerate(lst), lambda i_x: i_x[0] - i_x[1]):
+        group = list(map(lambda i_x: i_x[1], g))
+        if len(group) == 1:
+            ranges.append((group[0], group[0]))
+        else:
+            ranges.append((group[0],group[-1]))
+    return ranges
 
 def write_metadata():
+    """ Writes some metadata information to a metadata.yaml file. """
     uname = platform.uname()
     python_ver = 'python_ver: ' + platform.python_version()
     cuda_version = os.system("nvcc --version | sed -n 's/^.*release \([0-9]\+\.[0-9]\+\).*$/\1/p'")
```

Comment:
This is probably okay if you don't want to add a dependency, but [using cupy](https://docs.cupy.dev/en/latest/reference/generated/cupy.cuda.get_local_runtime_version.html) might be easier.